Index: motion-3.2.12/video.h
===================================================================
--- motion-3.2.12.orig/video.h	2011-08-23 15:42:35.564656448 -0430
+++ motion-3.2.12/video.h	2011-08-23 15:43:10.860657024 -0430
@@ -12,7 +12,7 @@
 
 #define _LINUX_TIME_H 1
 #ifndef WITHOUT_V4L
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <sys/mman.h>
 #include "pwc-ioctl.h"
 #endif
Index: motion-3.2.12/motion.h
===================================================================
--- motion-3.2.12.orig/motion.h	2011-08-23 15:44:46.108655912 -0430
+++ motion-3.2.12/motion.h	2011-08-23 15:45:21.240655678 -0430
@@ -40,7 +40,8 @@
 
 #define _LINUX_TIME_H 1
 #if !defined(WITHOUT_V4L) && !defined(BSD)
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
+#include <libv4l1-videodev.h>
 #endif
 
 #include <pthread.h>
