Description: Autodetect multiarch libavcodec path in autoconf.
Author: Stephen Kitt <steve@sk2.org>
Last-Update: 2012-02-12

--- a/configure.in
+++ b/configure.in
@@ -318,7 +318,12 @@
 	# AUTODETECT STATIC/SHARED LIB 
 	AC_MSG_CHECKING(for ffmpeg autodetecting libraries)
 
-	if test -f /usr/lib64/libavcodec.a -o -f /usr/lib64/libavcodec.so && test -f /usr/lib64/libavformat.a -o -f /usr/lib64/libavformat.so ; then
+	if test -f /usr/lib/$(dpkg-architecture -qDEB_BUILD_MULTIARCH)/libavcodec.so ; then
+		AC_MSG_RESULT(found in /usr/lib/$(dpkg-architecture -qDEB_BUILD_MULTIARCH))
+		FFMPEG_OK="found"
+		FFMPEG_LIB="/usr/lib/$(dpkg-architecture -qDEB_BUILD_MULTIARCH)"
+		FFMPEG_DIR="/usr"
+	elif test -f /usr/lib64/libavcodec.a -o -f /usr/lib64/libavcodec.so && test -f /usr/lib64/libavformat.a -o -f /usr/lib64/libavformat.so ; then
 		AC_MSG_RESULT(found in /usr/lib64)
 		FFMPEG_OK="found"
 		FFMPEG_LIB="/usr/lib64"
